### DISCUSSION TITLE
Problem using input() in a terminal worker

### DISCUSSION THREAD

**Discussion by Bot1-Pzon** [pyscript/pyscript#2055] [Q&A]
I was trying recreating one of my python projects using the standard **terminal input/output** system, but the output was this:

![image](https://github.com/pyscript/pyscript/assets/161156186/c600c360-6ff5-41e7-945d-ed07ef1d4160)

I traced the problem to the `input("")` commands in my code.

I thought terminal workers with `input()` were supported, I even found this example on the [official documentation](https://docs.pyscript.net/2024.5.1/user-guide/terminal/):
```
<script type="py" terminal worker>
    name = input("What is your name? ")
    print(f"Hello, {name}")
</script>
```
At first I thought it was a sort of fixable **thread** problem as shone [hear](https://docs.pyscript.net/2024.5.1/user-guide/workers/), but that same page states: 
> For Atomics to work you must ensure your web server enables the following headers (this is the default behaviour for [pyscript.com](https://pyscript.com/)):

so I tried to do it on **pyscript.com** but whit the same results.

**Comment by WebReflection** [comment]
I am not sure I am following ... when in doubt, check this PyTerminal demo: https://webreflection.github.io/coincident/test/pyterminal/

input is supported there with both MicroPython and Pyodide ... so I don't know what you are doing but you do need a `worker` attribute and correct headers to make it work.

Otherwise, please provide the minimal broken example you can present as HTML where `input` fails but I am afraid we have no issue with it right now.

P.S. in short, an empty screenshot doesn't really help us helping you ... questions like "*which PyScript version?*" or "*where is your HTML?*" and so on are inevitable here.

**Comment by JeffersGlass** [comment]
I tried your simple example both locally and on PyScript.com, and it seems to work fine for me on both:
![image](https://github.com/pyscript/pyscript/assets/1931111/bbd9aab0-9ab8-4e13-b600-f8c4a5bee418)

If you [open the developer console](https://screenful.com/guide/how-to/how-to-open-the-browser-developer-console), do you see any error messages?

**Comment by mhoffman02** [comment]
The May 11 code from Bot1-Pzon  , when run from pyscript.com, runs fine for "View in site" but errors out for "Preview" -- 
[Sample site ](https://_mike_.pyscriptapps.com/py-essentails-base/latest/)




`<!doctype html>

<html>
    <head>
        
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width,initial-scale=1.0">

        
         <link rel="stylesheet" href="https://pyscript.net/releases/2024.10.2/core.css">
        
        <link rel="stylesheet" href="./assets/css/examples.css" />
     
        
         <script type="module" src="https://pyscript.net/releases/2024.10.2/core.js"></script>
        
        <style>
            #loading { outline: none; border: none; background: transparent }
        </style>
        <script id="loader" type="module">
            //const loading = document.getElementById('loading');
            //addEventListener('py:ready', () => loading.close());
            //loading.showModal();
        </script>

        <title>Py Essentials</title>
        <link rel="icon" type="image/png" href="./assets/favicon.png" />
    </head>

    <body>
        <dialog id="loading">
            <h1>Loading...</h1>
        </dialog>
        
     <script type="py" terminal worker>
        name = input("What is your name? ")
        print(f"Hello, {name}")
    </script>
        
    </body>
</html>
`

What is your name? Traceback (most recent call last):
  File "/lib/python312.zip/_pyodide/_base.py", line 597, in eval_code_async
    await CodeRunner(
  File "/lib/python312.zip/_pyodide/_base.py", line 411, in run_async
    coroutine = eval(self.code, globals, locals)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<exec>", line 1, in <module>
OSError: [Errno 29] I/O error
